### PR TITLE
Api

### PR DESCRIPTION
**Pull request by EugeneLu12**

### CODE REVIEW

File: app/static/js/submitform.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+function make_request() { 
+    let queryField = document.getElementById('queryField').value;
+    let limitField = document.getElementById('limitField').value;
+    const http = new XMLHttpRequest();
+    const queryUrl = encodeURIComponent(`${queryField}`)
```
Comment (tso) [inline app/static/js/submitform.js]: Can ``${queryField}`` just be `queryField`? Same question with `limitField`

File: app/static/js/submitform.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    const limitUrl = encodeURIComponent(`${limitField}`)
+    url = `${window.location.origin}/api/player?query=${queryUrl}&limit=${limitUrl}`
+    http.open('GET', url);
```
Comment (tso) [inline app/static/js/submitform.js]: What if we used fetch instead? https://developers.google.com/web/updates/2015/03/introduction-to-fetch
```
fetch(url).then(function(response) {
  parsedResponseJSON = JSON.stringify(response.json()), undefined, 2);
  document.getElementById('playerResponse').innerHTML = parsedResponseJSON;
}
```

File: app/templates/about.html [hunk 3]
```diff
@@ -6,12 +6,65 @@
...
-      All inquiries can be sent to <i>YenFu#4967</i> via private message on <i>Discord</i>.
-    </h1>
+  <div class='content'>
+    <h1>About</h1>
+    <p>
+      sc2Ladder is a web app that displays the in-game ladders of StarCraft II and updates them in real time.
+      It also has the unique feature of revealing an account's Battle Net ID, revealing the identity of some barcode accounts.<br><br>
+      The source code can be found <a href="https://github.com/YenFu1/sc2ladder">here.</a><br><br>
+      All inquiries can be sent to <i>YenFu#4967</i> via private message on <i>Discord</i>.<br><br>
+      Special thanks to <a href="https://github.com/tso">tso</a>, <a href="https://github.com/jtleague9">woopr</a>, and
```
Comment (tso) [inline app/templates/about.html:16]: 😮

File: app/templates/about.html [hunk 4]
```diff
@@ -6,12 +6,65 @@
...
+        All endpoints limit requests to 25 results returned by default but the limit can be modified to return up to 200 results.
+      </p>
+  </div>
+  <div class="content">
+    <h1>Player API</h1>
+    <p>
+      Returns players whose character name or battletag matches the query parameter.<br><br>
+      <b>Request URL Format:</b> /api/player?query={query}
+    </p>
+    <form action="" method="get">
```
Comment (tso) [inline app/templates/about.html]: I think we remove `<form action="" method="get">`, and the `name=` on `queryField` and `limitField`.

File: app/views.py [hunk 5]
```diff
@@ -1,12 +1,26 @@
...
+from django.core.serializers.json import DjangoJSONEncoder
 
-from app.models import Player, Rank
+from app.models import Player, Rank, Race
 
-def index(request):
-    return render(request, 'index.html')
+class EnumEncoder(DjangoJSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, (Rank, Race)):
```
Comment (tso) [inline app/views.py:11]: Maybe
```
        if isinstance(obj, (Rank, Race)):
            return str(obj)
        return super().default(obj)
```

File: app/views.py [hunk 6]
```diff
@@ -1,12 +1,26 @@
...
-def search(request):
+
+def api_search(request):
+    players = list(
+        get_players(request).values('realm', 'region', 'rank', 'username', 'bnet_id', 'race',
+                                    'mmr', 'wins', 'losses', 'clan', 'profile_id'))
+    return JsonResponse(players, safe=False, encoder=EnumEncoder)
+
+
+def get_players(request):
```
Comment (tso) [inline app/views.py:23]: That's cute using request here. Not sure if this is the best idea but I'm down with it

**Review by tso — COMMENTED** [review]
hype PR

File: app/static/js/submitform.js (hunk 1 above)
Comment (EugeneLu12) [reply app/static/js/submitform.js]: yeah

File: app/views.py (hunk 5 above)
Comment (EugeneLu12) [reply app/views.py:11]: fixed

File: app/templates/about.html (hunk 4 above)
Comment (EugeneLu12) [reply app/templates/about.html]: fixed

File: app/static/js/submitform.js (hunk 2 above)
Comment (EugeneLu12) [reply app/static/js/submitform.js]: done

**Review by tso — APPROVED** [review]
hype

**Review by WilliamStewart — APPROVED** [review]

**Pull request merged**